nixos/modules/tasks/filesystems.nix

```diff
@@ -155,6 +174,17 @@ in
       description = "Names of supported filesystem types.";
     };
 
+    boot.specialFileSystems = mkOption {
+      type = types.listOf (types.submodule coreFileSystemOpts);
+      apply = makeSpecialMounts;
```

Comment:
I thought about the fstab format as well, but that probably won't work too well since someone has to create the mountpoint directories beforehand (and busybox doesn't seem to support `x-mount.mkdir[=mode]` like util-linux).